### PR TITLE
fix `unexpected end of JSON input` during planning

### LINKED ISSUE #219: json error during kustomizationResourceDiff
We have a terraform config like

```
# Other modules and terraform
... 
module "foo" {
  helm_version = "1.2.0"
...
}

module "bar" {
  # this module uses the kbst provider
  depends_on [
    module.foo.release_metdata
  ]
}
```

The bar module, which installs fluxcd.io, does not use the any output of the foo module, it just has a dependency on 1 output to ensure that the foo module completes 1st.  Now when the terraform config is 1st created all is well.  And all subsequent plans are fine too, until I update the version # in the foo module to say 1.2.1.  Then the terraform plan will always fails with errors like 
```
╷
│ Error: github.com/kbst/terraform-provider-kustomize/kustomize.kustomizationResourceDiff: github.com/kbst/terraform-provider-kustomize/kustomize.(*kManifest).load: json error: unexpected end of JSON input
│ 
│   with module.cpu_dev2_eks.module.eks_cluster_init.module.fluxcd.kustomization_resource.p0["_/Namespace/_/flux-system"],
│   on .terraform/modules/cpu_dev2_eks.eks_cluster_init.fluxcd/fluxcd/main.tf line 59, in resource "kustomization_resource" "p0":
│   59: resource "kustomization_resource" "p0" {
│ 
╵
╷
│ Error: github.com/kbst/terraform-provider-kustomize/kustomize.kustomizationResourceDiff: github.com/kbst/terraform-provider-kustomize/kustomize.(*kManifest).load: json error: unexpected end of JSON input
│ 
│   with module.cpu_dev2_eks.module.eks_cluster_init.module.fluxcd.kustomization_resource.p0["apiextensions.k8s.io/CustomResourceDefinition/_/providers.notification.toolkit.fluxcd.io"],
│   on .terraform/modules/cpu_dev2_eks.eks_cluster_init.fluxcd/fluxcd/main.tf line 59, in resource "kustomization_resource" "p0":
│   59: resource "kustomization_resource" "p0" {
│ 
╵
╷
│ Error: github.com/kbst/terraform-provider-kustomize/kustomize.kustomizationResourceDiff: github.com/kbst/terraform-provider-kustomize/kustomize.(*kManifest).load: json error: unexpected end of JSON input
│ 
│   with module.cpu_dev2_eks.module.eks_cluster_init.module.fluxcd.kustomization_resource.p0["apiextensions.k8s.io/CustomResourceDefinition/_/alerts.notification.toolkit.fluxcd.io"],
│   on .terraform/modules/cpu_dev2_eks.eks_cluster_init.fluxcd/fluxcd/main.tf line 59, in resource "kustomization_resource" "p0":
│   59: resource "kustomization_resource" "p0" {
│ 
╵
...
```

This continues until I set the version in the foo module back to the original 1.2.0 and then the plans will successfully complete.

### PR DESCRIPTION
**Pull request by nazarewk**
github.com/kbst/terraform-provider-kustomize/kustomize.kustomizationResourceDiff: github.com/kbst/terraform-provider-kustomize/kustomize.(*kManifest).load: json error: unexpected end of JSON input

not sure if it's the right way to go, but my workday is now over and i still want to share the findings

fixes #219

### CODE REVIEW

**Comment by nazarewk** [comment]
I've tried refactoring to see if there would be any code left after IF-ing everything depending on new value, but there wasn't anything left except loading old manifest, so it's mostly just improved error message and returning early without errors.

File: kustomize/resource_kustomization.go
Comment (pst) [inline kustomize/resource_kustomization.go:191]: I will have to look into this to fully understand it before I can merge.

File: kustomize/resource_kustomization.go
Comment (pst) [inline kustomize/resource_kustomization.go:197]: You can not print the manifest to the log. The resource may be a secret, and now you printed that to the CI/CD log. Please remove this part.

**Review by pst — CHANGES_REQUESTED** [review]

File: kustomize/resource_kustomization.go
Comment (nazarewk) [reply kustomize/resource_kustomization.go:191]: You can quite easily replicate the issue by creating something like:
1. `resource "random_id"`
2. `data "kustomization_overlay" "prerender"`
3. `data "kustomization_overlay" "source"` - make this depend on the `random_id`
4. `resource "kustomization_resource"` depending on ids from `prerender` and manifests from `source`

**Comment by nazarewk** [comment]
I've noticed the same error happens after I delete last-applied config from a resource

**Comment by wpbeckwith** [comment]
What's the status on this issue?  I can see it's being worked, more so trying to get an idea if we need to actually do something or wait for an imminent fix.

**Comment by pst** [comment]
I finally found the time to look into this, and I unfortunately have to tell you that I have decided to not merge this.

I don't see any real world situation where this is an issue, your prerender "hack" to work around the `for_each` limitation with known after apply values is not a valid use in my opinion. If you depend on values that are known after apply, avoid the `for_each` by having one `kustomization_resource` per K8s resource. Or fall back to the `apply --target` as suggested by Terraform.

If there are valid uses where the provider crashes because of such an issue

**Pull request closed without merging**
